src/domain-services/flows/flow-service.ts

```diff
@@ -0,0 +1,544 @@
+import { type Database } from '@unocha/hpc-api-core/src/db';
+import { type FlowId } from '@unocha/hpc-api-core/src/db/models/flow';
+import { Op } from '@unocha/hpc-api-core/src/db/util/conditions';
+import { type InstanceOfModel } from '@unocha/hpc-api-core/src/db/util/types';
+import { createBrandedValue } from '@unocha/hpc-api-core/src/util/types';
+import { Service } from 'typedi';
+import { FlowObjectService } from '../flow-object/flow-object-service';
+import type {
+  FlowObject,
+  FlowObjectFilterGrouped,
+  FlowObjectType,
+} from '../flow-object/model';
+import { buildWhereConditionsForFlowObjectFilters } from '../flow-object/utils';
+import { type FlowParkedParentSource } from './graphql/types';
+import type {
+  FlowInstance,
+  FlowOrderBy,
+  FlowOrderByCond,
+  FlowWhere,
+  IGetFlowsArgs,
+  UniqueFlowEntity,
+} from './model';
+import { buildSearchFlowsConditions } from './strategy/impl/utils';
+
+@Service()
+export class FlowService {
+  constructor(private readonly flowObjectService: FlowObjectService) {}
+
+  async getFlows(args: IGetFlowsArgs): Promise<FlowInstance[]> {
+    const { models, orderBy, conditions, limit, offset } = args;
+
+    const distinctColumns = [
```

Comment:
Let's declare types here too.
```suggestion
    const distinctColumns: Array<keyof FlowInstance> = ['id', 'versionID'];
```